Encryption password

jasminewatissee opened the pull request.

jasminewatissee committed:
Class password

Creation of a class password

yassine-turki committed:
Generate salt

jasminewatissee committed:
Small corrections in random_character

yassine-turki committed:
Updates

yassine-turki committed:
Password

yassine-turki committed:
Password/ Username and Profile classes with

yassine-turki committed:
Create .gitignore

yassine-turki committed:
Merge branch 'main' into Encryption-Password

yassine-turki committed:
Merge

yassine-turki committed:
Update password.cpp

jasminewatissee committed:
Added algorithm to test the hash

jasminewatissee committed:
Merge branch 'Encryption-Password' of https://github.com/sebiinfo/Authentification-System into Encryption-Password

jasminewatissee committed:
Optimization and correction of bugs

jasminewatissee committed:
List of words to test the encryption function

sophieantoun committed:
Hash function first draft

yassine-turki committed:
Attempt to debug hash_it

jasminewatissee committed:
Merge branch 'Encryption-Password' of https://github.com/sebiinfo/Authentification-System into Encryption-Password

yassine-turki committed:
Update password.hpp

jasminewatissee committed:
Small update class profile

jasminewatissee committed:
Merge branch 'Encryption-Password' of https://github.com/sebiinfo/Authentification-System into Encryption-Password

yassine-turki committed:
Update

jasminewatissee committed:
New class brouillon

jasminewatissee committed:
Merge branch 'Encryption-Password' of https://github.com/sebiinfo/Authentification-System into Encryption-Password

yassine-turki committed:
Test function

sophieantoun committed:
Create database.cpp

sophieantoun committed:
Create database.hpp

jasminewatissee committed:
Simplification & optimization of the class

jasminewatissee committed:
Merge branch 'Encryption-Password' of https://github.com/sebiinfo/Authentification-System into Encryption-Password

jasminewatissee committed:
Debugging & testing of the class

jasminewatissee committed:
Update main (there are still problems for the test)

yassine-turki committed:
Database test(w dictionnary)

jasminewatissee committed:
Debugging of the testing function

jasminewatissee committed:
Test function works, prints to file

jasminewatissee committed:
testing works + removal of std namespace

jasminewatissee committed:
removal of "using std namespace"

jasminewatissee committed:
Comments

jasminewatissee committed:
Deletion of database test

moreiramn commented:
I just started reviewing the code, but a first remark that you can already quickly fix is to place your files under "sources/Password". Moreover, please do not include the build files in the Repo, as these are personal to each one's computer and will create a lot of merge conflicts.

jasminewatissee committed:
Directory changed

jasminewatissee committed:
Variable names to snake case

moreiramn commented:
On password.cpp and password.hpp there are a lot of lines that are currently commented. If they are not going to be used anymore, I would prefer them to be deleted. Otherwise, what is the intent of keeping them?

jasminewatissee committed:
Removal of comments

moreiramn commented:
In your Profile class, you generate a random salt per password. How can you guarantee that it will use the same salt when you re-hash the password?

moreiramn committed:
Relative Paths and proper includes in tests.cpp

jasminewatissee committed:
compare_password changed to use salt in database

jasminewatissee committed:
Implementation of validate_password that checks if the password answers certain conditions

moreiramn committed:
formatting of password

jasminewatissee committed:
changes in generate_salt

moreiramn committed:
swap sizeof() for .size()

yassine-turki committed:
Update Q number for Password branch

yassine-turki committed:
Cmake file

yassine-turki committed:
More cmake changes

yassine-turki committed:
Small changes in Password.cpp

yassine-turki committed:
Testing textfile1 for password

yassine-turki committed:
Changes in Password

yassine-turki committed:
Update tests.cpp

yassine-turki committed:
Update main.cpp

yassine-turki committed:
hash_it function updated

Updated function, switched from robin Karp's algorithm to an 8 byte version of MurmurHash 3 algorithm

yassine-turki committed:
Deletd comments and Q variable

moreiramn committed:
Untracking a few files

moreiramn committed:
Final Renaming to merge
